### PR TITLE
feat(lifecycle): add handling of a first representative for a resource

### PR DESCRIPTION
**Pull request by RadovanTomik**
### Description:

This PR adds functionality to update the resource status in old Negotiations when the resource was marked as unreachable.

### Checklist:

_Make sure you tick all the boxes bellow if they are true or do not apply:_

- [x] I have performed a self review of my code
- [x] My code follows [Google Java Code style](https://google.github.io/styleguide/javaguide.html)
- [x] I have made my code as simple as possible
- [x] I have added unit tests and the code coverage has not decreased
- [x] I have updated the documentation in all relevant places

### CODE REVIEW

File: src/main/java/eu/bbmri_eric/negotiator/database/model/Negotiation.java [hunk 1]
```diff
@@ -146,13 +140,16 @@ public void setCurrentState(NegotiationState negotiationState) {
   public void setStateForResource(String resourceId, NegotiationResourceState state) {
     currentStatePerResource.put(resourceId, state);
     if (!state.equals(NegotiationResourceState.SUBMITTED)) {
+      System.out.println("here");
```
Comment (konradlang) [inline src/main/java/eu/bbmri_eric/negotiator/database/model/Negotiation.java:143]: is there a reason to have the println in here?

File: src/main/java/eu/bbmri_eric/negotiator/database/model/Negotiation.java [hunk 2]
```diff
@@ -146,13 +140,16 @@ public void setCurrentState(NegotiationState negotiationState) {
...
       NegotiationResourceLifecycleRecord record =
           NegotiationResourceLifecycleRecord.builder()
+              .negotiation(this)
               .changedTo(state)
               .resource(lookupResource(getResources(), resourceId))
               .build();
       this.negotiationResourceLifecycleRecords.add(record);
     }
+    System.out.println(this.negotiationResourceLifecycleRecords.size());
```
Comment (konradlang) [inline src/main/java/eu/bbmri_eric/negotiator/database/model/Negotiation.java:152]: If this is for logging purpose please add this to the appropriate log level.

File: src/main/java/eu/bbmri_eric/negotiator/database/model/Negotiation.java (hunk 1 above)
Comment (RadovanTomik) [reply src/main/java/eu/bbmri_eric/negotiator/database/model/Negotiation.java]: sorry leftovers from debugging

**Review by konradlang — APPROVED** [review]

**Comment by RadovanTomik** [comment]
add tests for event publishing and making sure this functionality doesn't update the state of other collections

**Pull request merged**